### DISCUSSION TITLE
outdated(?) image component properties in next.js

### DISCUSSION THREAD

**Discussion by aw-1224rst** [neuefische/questions#320] [Web] [answered]
I'm using the properties "objectFit" and "layout" in my next.js image component to be able to display a certain part of a picture without distortion.
The pictures get displayed properly, but my console is running these warnings:

> Image with src "/img/eventhorizon.jpeg" has "fill" but is missing "sizes" prop. Please add it to improve page performance.
i get this warning for all images.

> warn-once.js?7395:11 Image with src "/img/firefly.jpeg" was detected as the Largest Contentful Paint (LCP). Please add the "priority" property if this image is above the fold.

> warn-once.js?7395:11 Image with src "/img/midsommar.jpeg" has legacy prop "objectFit". Did you forget to run the codemod?

[pull request](https://capstone-git-display-feature-wilhelmalfke.vercel.app/)

` <ListItem key={movie.id}>
            <Image
              layout="fill"
              objectFit="cover"
              alt="picture of movie poster"
              src={movie.picture}
            />
          </ListItem>`
(code from components/ListItemSquare/index.js, line: 9-18  
heres a link to my app, however there the warning doesnt apear...
[capstone preview](https://capstone-git-display-feature-wilhelmalfke.vercel.app/)

**Comment by JessicaLoers** [comment, marked as answer]
Hi @WilhelmAlfke 
That's the reason why you get these warnings 👇
> Starting with  Next.js 13, the `next/image` component was rewritten [...] the old `next/image` was renamed to `next/lagacy/image` ([see Next.js documentation Link](https://nextjs.org/docs/api-reference/next/legacy/image))

Next.js 13 `next/image` component doesn't supports the following properties any longer: 
`layout, objectFit, objectPosition`


You have two ✌️ options:
**1. Change your import** 
`import Image from 'next/legacy/image`';

**2. Use css instead, therefor you need to set the height and width properties ..  and the priority property as well I guess 🫣**
`import Image from 'next/image`';

```js
import styled from 'styled-components';
import Image from 'next/image';

export default function ListItemSquare({ movies }) {
  return (
      {movies.map((movie) => {
        return (
          <ListItem key={movie.id}>
            <StyledImage
              height={100}
              width={100}
              priority
              alt={`movieposter of ${movie.name}`}
              src={movie.picture}
            />
          </ListItem>
        );
      })}
  );
}

const StyledImage = styled(Image)`
  object-fit: cover;
`;

const ListItem = styled.li`
// ... 
`;
```
